src/controllers/achievementsController.ts

```diff
@@ -64,24 +64,29 @@ export class AchievementsController implements AchievementsControllerInterface {
 	public async getVolunteersPage(req: Request, res: Response, next: NextFunction) {
 		try {
 			let users: User[] = await this._userService.getAllUsers();
-			users = users.sort((a: User, b: User) => a.getName().localeCompare(b.getName()));
+			users = users.sort((a: User, b: User) => a.name.localeCompare(b.name));
+
+			const usersMap: Map<string, User> = new Map();
+			users.forEach((user: User) => {
+				usersMap.set(user.id, user);
+			});
 
 			let achievements: Achievement[] = await this._achievementsService.getAchievements();
 			achievements = achievements.sort((a: Achievement, b: Achievement) => a.getTitle().localeCompare(b.getTitle()));
 
 			let prizesToClaim: AchievementProgress[] = await this._achievementsProgressService.getAchievementsProgressThatCanClaimPrize();
 			prizesToClaim = prizesToClaim.sort((a: AchievementProgress, b: AchievementProgress) => {
 				const achievementsComparison: number = a.getAchievement().getTitle().localeCompare(b.getAchievement().getTitle());
-				if (achievementsComparison === 0) {
-					return a.getUser().getName().localeCompare(b.getUser().getName());
+				if (achievementsComparison === 0 && usersMap.get(a.userId) !== undefined && usersMap.get(b.userId) !== undefined) {
```

Comment:
```suggestion
				if (achievementsComparison === 0 && typeof usersMap.get(a.userId) !== 'undefined' && typeof usersMap.get(b.userId) !== 'undefined') {
```

I think this is the safer way to do undefined checking in JS

Reply:
I think that it true in JavaScript but in typescript it's okay to do !== undefined and the TSC will convert it to the safe JS check